### PR TITLE
ipatests: Tests for RBCD feature

### PR DESCRIPTION
**Pull request by miskopo**
Integration tests for RBCD feature.

Related: xxx

### CODE REVIEW

**Comment by miskopo** [comment]
@abbra could you please take a look at this test proposal?

File: ipatests/test_integration/test_rbcd.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+        cls.ad = cls.ads[0]
+        cls.ipa-client = cls.clients[0]
+        cls.ad_user = '{}@{}'.format(cls.ad_user_login, cls.ad.domain.name)
+
+        tasks.install_adtrust(cls.master)
+        tasks.configure_dns_for_trust(cls.master, cls.ad)
+        tasks.configure_windows_dns_for_trust(cls.ad, cls.master)
+        tasks.establish_trust_with_ad(cls.master, cls.ad.domain.name,
+                                      extra_args=['--two-way=false'])
```
Comment (abbra) [inline ipatests/test_integration/test_rbcd.py:53]: You must use `--two-way=true` for any cross-realm S4U scenarios. Please see https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-sfu/1fb9caca-449f-4183-8f7a-1a5fc7e7290a and https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-sfu/f35b6902-6f5e-4cd0-be64-c50bbaaf54a5 for more details.

File: ipatests/test_integration/test_rbcd.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        ipa_service = "host" + "/" + self.master.hostname
+
+        cmd = ('New-ADComputer -Name {} -AccountPassword (ConvertTo-SecureString -String {} -Force -AsPlainText) '
+               '-ServicePrincipalNames {} -CannotChangePassword $true -PasswordNeverExpires $true '
+               '-KerberosEncryptionType AES128,AES256 -TrustedForDelegation $true '
+               '-Enabled $true -AccountNotDelegated $false').format(service_name, service_paswd, self.ad_domain)
+        self.ad.run_command(['powershell', '-c', cmd])
+
+        #Set RBCD
+        cmd = ('Set-ADComputer -Identity {} -PrincipalsAllowedToDelegateToAccount '{}).format(service_name, ipa_service)
```
Comment (abbra) [inline ipatests/test_integration/test_rbcd.py:75]: This is the part that will not work currently. Powershell commands cannot resolve IPA services in any way and `Set-ADComputer` cannot take SID of IPA service. I am still investigating how we can define that list...

**Review by abbra — CHANGES_REQUESTED** [review]
I left few comments.

**Pull request closed without merging**